CMakeLists.txt

```diff
@@ -13,6 +13,8 @@ TRIBITS_PACKAGE_DECL(DataTransferKit)
 TRIBITS_ADD_DEBUG_OPTION()
 TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+
```

Comment:
Well, if I understand CMake correctly, it will only append `-Werror` while building DTK code, and not otherwise. I may be wrong about it.